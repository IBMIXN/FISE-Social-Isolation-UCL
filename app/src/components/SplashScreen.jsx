import React from "react";
import { Flex, Stack, Icon, Text, Spinner } from "@chakra-ui/core";

const SplashScreen = () => (
  <Flex
    direction="column"
    position="fixed"
    justify="center"
    align="center"
    backgroundColor="gray.800"
    backgroundSize="cover"
    minWidth="100%"
    minHeight="100%"
    backgroundPosition="center"
    top="0"
    left="0"
  >
    <Spinner
      size="xl"
      color="blue.500"
      mt="2rem"
      thickness="5px"
      speed="0.6s"
    />

    <Text p="1rem " textAlign="center" color="white">
      FISE 2021 by UCL and IBM
    </Text>
    <Stack>
      <Icon name="ibm" color="white" w="200px" h="50px" m="1rem" />
      <Icon name="nhsX" color="white" w="200px" h="60px" m="1rem" />
      <Icon name="ucl" color="gray.400" w="200px" h="50px" m="1rem" />
    </Stack>

    <Stack p="1rem" color="gray.300">
      <Text textAlign="center">
        In association with IBM, NHS & University College London
      </Text>
      <Text
        textAlign="center"
        maxWidth="35em"
        color="gray.500"
        wordBreak="true"
      >
        The software is an early proof of concept for development purposes and
        should not be used as-is in a live environment without further
        redevelopment and/or testing. No warranty is given and no real data or
        personally identifiable data should be stored. Usage and its liabilities
        are your own.
      </Text>
    </Stack>
  </Flex>
);

export default SplashScreen;
