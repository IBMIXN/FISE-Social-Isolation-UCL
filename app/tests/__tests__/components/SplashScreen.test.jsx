import React from 'react';
import ReactDOM from 'react-dom';
import SplashScreen from '../../../src/components/SplashScreen';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('render unit test', () => {
  const wrapper = shallow(<SplashScreen/>)
  
  expect(toJson(wrapper)).toMatchSnapshot();
});

// it('integration test', () => {
//   const wrapper = mount(<SplashScreen />)

//   expect(toJson(wrapper)).toMatchSnapshot();
// });