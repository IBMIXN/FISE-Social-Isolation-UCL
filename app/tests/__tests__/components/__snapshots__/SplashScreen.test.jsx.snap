// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render unit test 1`] = `
<Flex
  align="center"
  backgroundColor="gray.800"
  backgroundPosition="center"
  backgroundSize="cover"
  direction="column"
  justify="center"
  left="0"
  minHeight="100%"
  minWidth="100%"
  position="fixed"
  top="0"
>
  <Text
    color="white"
    textAlign="center"
  >
    by UCL FISE Groups
  </Text>
  <ForwardRef>
    <Icon
      color="white"
      h="50px"
      m="1rem"
      name="ibm"
      w="200px"
    />
    <a
      href="https://nhs.uk"
    >
      <Image
        h="81px"
        mb="1.2rem"
        mt="1.2rem"
        mx="auto"
        src="/icons/nhs-inx-logo.png"
        w="157px"
      />
    </a>
    <Icon
      color="gray.400"
      h="50px"
      m="1rem"
      name="ucl"
      w="200px"
    />
  </ForwardRef>
  <Spinner
    color="blue.500"
    mt="2rem"
    size="xl"
    speed="0.6s"
    thickness="5px"
  />
  <ForwardRef
    color="gray.300"
    p="1rem"
  >
    <Text
      textAlign="center"
    >
      In association with IBM, NHS & University College London
    </Text>
    <Text
      textAlign="center"
    >
      Supervised by John McNamara, Dean Mohamedally
    </Text>
  </ForwardRef>
</Flex>
`;
